src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -482,9 +485,139 @@ EventEngine::TaskHandle PosixEventEngine::RunAfterInternal(
   return handle;
 }
 
+PosixEventEngine::PosixDNSResolver::PosixDNSResolver(
+    ResolverOptions const& options, PosixEnginePollerManager* poller_manager,
+    PosixEventEngine* event_engine)
+    : options_(options),
+      poller_manager_(poller_manager),
+      event_engine_(event_engine) {}
+
+PosixEventEngine::PosixDNSResolver::~PosixDNSResolver() {}
+
+using LookupTaskHandle = PosixEventEngine::PosixDNSResolver::LookupTaskHandle;
```

Comment:
`using` statements should be grouped at the top of the inner-most namespace scope. See https://google.github.io/styleguide/cppguide.html#Declaration_Order